dockstore-integration-testing/src/test/java/io/dockstore/client/cli/RefreshByOrgIT.java

```diff
@@ -0,0 +1,203 @@
+    private static Client client;
+    private static String token;
+    private static String usersURLPrefix;
+    private static ObjectMapper objectMapper;
+    private static Long id;
+
+    @BeforeClass
+    public static void clearDBandSetup() throws IOException, TimeoutException, ApiException {
+        clearStateMakePrivate2();
+        final CommonTestUtilities.TestingPostgres testingPostgres = getTestingPostgres();
+        id = testingPostgres.runSelectStatement("select id from enduser where username='DockstoreTestUser2';", new ScalarHandler<>());
+        Environment environment = RULE.getEnvironment();
+        token = testingPostgres.runSelectStatement("select content from token where tokensource='dockstore';", new ScalarHandler<>());
+        client = new JerseyClientBuilder(environment).build("test client").property(ClientProperties.READ_TIMEOUT, 60000);
+        objectMapper = environment.getObjectMapper();
+    }
+
+    @Before
+    public void clearDB() throws IOException, TimeoutException, ApiException {
+        clearStateMakePrivate2();
+    }
+
+    private void checkInitialDB() throws IOException {
+        // The DB initially has no workflows
+        assertThat(getWorkflows().size()).isGreaterThanOrEqualTo(0);
+        // The DB initially has 4 tools, 1 from dockstore2 and 3 from dockstoretestuser2
+        assertThat(getTools().size()).isGreaterThanOrEqualTo(4);
+    }
+
+    @Test
+    public void testRefreshToolsByOrg() throws IOException {
+        usersURLPrefix = "http://localhost:%d/users/" + id;
+        checkInitialDB();
+        testRefreshToolsByOrg1();
+        assertThat(getTools().size()).isGreaterThanOrEqualTo(4);
+        testRefreshToolsByOrg2();
+        assertThat(getTools().size()).isGreaterThanOrEqualTo(5);
+        testRefreshToolsByOrg3();
+        assertThat(getTools().size()).isGreaterThanOrEqualTo(5);
+    }
+
+    /**
+     * This tests if refreshing dockstore2 retains the same number of repos
+     */
+    private void testRefreshToolsByOrg1() throws IOException {
+        String url = usersURLPrefix + "/containers/dockstore2/refresh";
+        List<Tool> tools = clientHelperTool(url);
+        assertThat(tools.size()).isGreaterThanOrEqualTo(4);
+    }
+
+    /**
+     * This tests if refreshing dockstorestestuser2 adds 1 additional repo
+     */
+    private void testRefreshToolsByOrg2() throws IOException {
+        String url = usersURLPrefix + "/containers/dockstoretestuser2/refresh";
+        List<Tool> tools = clientHelperTool(url);
+        assertThat(tools.size()).isGreaterThanOrEqualTo(5);
+
+    }
+
+    /**
+     * This tests if refreshing a non-existent repo does not alter the current 5 repos
+     */
+    private void testRefreshToolsByOrg3() throws IOException {
+        String url = usersURLPrefix + "/containers/mmmrrrggglll/refresh";
+        List<Tool> tools = clientHelperTool(url);
+        assertThat(tools.size()).isGreaterThanOrEqualTo(5);
```

Comment:
in this case, just record the number of tools before the refresh and compare afterwards. Otherwise, this test is in danger of always passing if we add a large number of tools to the DB for other reasons